pkgs/development/libraries/libproxy/default.nix

```diff
@@ -1,82 +1,77 @@
 { lib
-, stdenv
+, bash
+, curl
+, duktape
+, enableDuktape ? lib.meta.availableOn stdenv.hostPlatform duktape
 , fetchFromGitHub
 , fetchpatch
-, pkg-config
-, cmake
-, zlib
-, dbus
-, networkmanager
-, enableJavaScript ? stdenv.isDarwin || lib.meta.availableOn stdenv.hostPlatform duktape
-, duktape
-, pcre
-, gsettings-desktop-schemas
+, gi-docgen
 , glib
-, makeWrapper
-, python3
-, SystemConfiguration
+, gobject-introspection
+, gsettings-desktop-schemas
+, meson
+, ninja
+, pkg-config
+, stdenv
+, vala
 , CoreFoundation
 , JavaScriptCore
+, SystemConfiguration
 }:
 
 stdenv.mkDerivation rec {
   pname = "libproxy";
-  version = "0.4.18";
+  version = "0.5.3";
+
+  outputs = [ "out" "dev" ];
 
   src = fetchFromGitHub {
     owner = "libproxy";
     repo = "libproxy";
     rev = version;
-    hash = "sha256-pqj1LwRdOK2CUu3hYIsogQIXxWzShDuKEbDTbtWkgnQ=";
+    hash = "sha256-qdYB6HJkgboS8kkTvTqLy6Z3JYY5SOJsRl6nZM0iuvw=";
   };
 
-  patches = lib.optionals stdenv.isDarwin [
-    # https://github.com/libproxy/libproxy/pull/189
+  patches = [
     (fetchpatch {
-      url = "https://github.com/libproxy/libproxy/commit/4331b9db427ce2c25ff5eeb597bec4bc35ed1a0b.patch";
-      sha256 = "sha256-uTh3rYVvEke1iWVHsT3Zj2H1F+gyLrffcmyt0JEKaCA=";
+      url = "https://github.com/libproxy/libproxy/pull/261/commits/fe64d1f56b7695bbc8ccb31a34546bc289a4c44d.patch";
+      hash = "sha256-X13rXtmWvssdxy795Ag+2op1vUpEqt6+aicR64zKQnQ=";
     })
   ];
 
-  outputs = [ "out" "dev" "py3" ];
+  postPatch = ''
+    # prevent meson from installing git hooks...
+    # https://github.com/libproxy/libproxy/issues/262
+    echo -e "#!${lib.getExe bash}\nexit 0" > data/install-git-hook.sh
+    chmod +x data/install-git-hook.sh
+  '';
 
   nativeBuildInputs = [
+    gi-docgen
+    gobject-introspection
+    meson
+    ninja
     pkg-config
-    cmake
-    makeWrapper
+    vala
   ];
 
   buildInputs = [
-    pcre
-    python3
-    zlib
-  ] ++ lib.optionals enableJavaScript [
-    (if stdenv.hostPlatform.isDarwin then JavaScriptCore else duktape)
+    curl
+  ] ++ lib.optionals enableDuktape [
+    duktape
   ] ++ (if stdenv.hostPlatform.isDarwin then [
-    SystemConfiguration
     CoreFoundation
+    JavaScriptCore
+    SystemConfiguration
   ] else [
     glib
-    dbus
-    networkmanager
+    gsettings-desktop-schemas
   ]);
 
-  cmakeFlags = [
-    "-DWITH_PYTHON2=OFF"
-    "-DPYTHON3_SITEPKG_DIR=${placeholder "py3"}/${python3.sitePackages}"
-  ] ++ lib.optional (enableJavaScript && !stdenv.hostPlatform.isDarwin) "-DWITH_MOZJS=ON";
-
-  postFixup = lib.optionalString stdenv.isLinux ''
-    # config_gnome3 uses the helper to find GNOME proxy settings
-    wrapProgram $out/libexec/pxgsettings --prefix XDG_DATA_DIRS : "${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}"
```

Comment:
If this still looks for GNOME settings, either a wrapper or patch (see `makeHardcodeGsettingsPatch`) will still be necessary.

Reply:
pxgsettings no longer exists.

Reply:
😒 https://github.com/libproxy/libproxy/commit/3e06882f03afd7020da050902169b9e81fa23163

The new location seems to be https://github.com/libproxy/libproxy/tree/main/src/backend/plugins/config-gnome .

No idea if we need to wrap the proxy binary instead.

Reply:
`proxy` program is [just a test tool](https://github.com/libproxy/libproxy/blob/42b58efd268bc0726dd4ec4e6a37d6ebb5e09cb5/src/tools/proxy.8#L5) so wrapping it will not help. Looks like the backends are all compiled into libpxbackend.

The only option is patching it as in https://github.com/NixOS/nixpkgs/pull/271037

Reply:
So, do we now wait for that PR or just hope people wrap?

Reply:
That PR only affects GTK. I meant is as an example what would have to be done here if we do not want even more wrappers everywhere.

Reply:
I have no experience with C/C++, so that would need to be done by someone else.

Reply:
I have pushed a patch + update script to update it (depends on https://github.com/NixOS/nixpkgs/pull/276828). The patch is suboptimal for now (the source is created multiple times) but at least it can be generated automatically.